From 83efe732d6bba81d636fdda4ace3c211f37e5c61 Mon Sep 17 00:00:00 2001
From: Gianluca Gennari <gennarone@gmail.com>
Date: Fri, 23 Mar 2018 14:01:50 +0100
Subject: [PATCH] it913x: switch off PID filter by default

PID filter can cause troubles, as reported by a few users, so switch it off
by default as we do not need it anyway.

---
 drivers/media/usb/dvb-usb-v2/it913x.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/media/usb/dvb-usb-v2/it913x.c b/drivers/media/usb/dvb-usb-v2/it913x.c
index fe95a586..3c726dec 100644
--- a/drivers/media/usb/dvb-usb-v2/it913x.c
+++ b/drivers/media/usb/dvb-usb-v2/it913x.c
@@ -43,6 +43,10 @@ static int dvb_usb_it913x_debug;
 module_param_named(debug, dvb_usb_it913x_debug, int, 0644);
 MODULE_PARM_DESC(debug, "set debugging level (1=info (or-able)).");
 
+static int disable_pid_filter = 1;
+module_param_named(pid, disable_pid_filter, int, 0644);
+MODULE_PARM_DESC(pid, "set 0=on default 1=off");
+
 static int dvb_usb_it913x_firmware;
 module_param_named(firmware, dvb_usb_it913x_firmware, int, 0644);
 MODULE_PARM_DESC(firmware, "set firmware 0=auto "\
@@ -419,6 +423,12 @@ static int it913x_get_stream_config(struct dvb_frontend *fe, u8 *ts_type,
 		struct usb_data_stream_properties *stream)
 {
 	struct dvb_usb_adapter *adap = fe_to_adap(fe);
+
+	if (disable_pid_filter == 1) {
+		/* module param: disable pid filtering */
+		adap->pid_filtering = 0;
+	}
+
 	if (adap->pid_filtering)
 		stream->u.bulk.buffersize = TS_BUFFER_SIZE_PID;
 	else
